File: proxy-agent/src/main/java/com/epam/drill/agent/proxy/ProxyAgent.java
package com.epam.drill.agent.proxy;

import com.alibaba.ttl.threadpool.agent.TtlAgent;

import java.lang.instrument.Instrumentation;

class ProxyAgent {
    public static void premain(String agentArgs, Instrumentation instrumentation) {
        TtlAgent.premain(agentArgs, instrumentation);
        System.out.println(">>>Ttl agent attached.");
    }
}

File: proxy-agent/src/main/kotlin/com.epam.drill.agent/ProxyAgent.kt
package com.epam.drill.agent

import com.alibaba.ttl.threadpool.agent.*
import java.lang.instrument.*

fun premain(args: String?, instrumentation: Instrumentation) {
    TtlAgent.premain(args ?: "", instrumentation)
    println("Ttl agent is attached")
}